### PR TITLE
Docs: File Library

### PR DESCRIPTION
**Pull request by erondpowell**
**TO DO:**
- [x] How it Works
- [x] Files
- [x] Editing an Image
- [x] Upload a File
- [x] Replace a File
- [x] Folders
- [x] Create a Folder
- [x] Rename a Folder
- [x] Move a Folder
- [x] Delete a Folder
- [x] Extensibility Options
- [x] More Help
- [x] Redo ALL Media: waiting for 2 videos until the cloud update
- [x] Proofread and touchup copy

**Questions:** (**CC** @rijkvanzanten)
- [x] Limits and controls for file sizes?
- [x] Are there *any* file types that cannot be imported?
- [x] Is it feasible to add an extension to upload Folders?
- [x] Is cropping reversible? Or is this a destructive change? Idk how to reverse it!

### CODE REVIEW

File: docs/app/file-library.md [hunk 1]
```diff
@@ -1,137 +1,219 @@
...
+- [Extensions > Creating Extensions](/extensions/creating-extensions/)
+- [Contributing > Introduction](/contributing/introduction/)
+- [Contributing > Codebase Overview](/contributing/codebase-overview/)
+
+::: tip Accelerated Development
+
+Working on an enterprise project and looking to outsource or financially sponsor the development of a Shares extension?
+Contact [our team](https://directus.io/contact/)
+
+:::
```
Comment (rijkvanzanten) [inline docs/app/file-library.md:214]: Lets remove this

File: docs/app/file-library.md [hunk 2]
```diff
@@ -1,137 +1,219 @@
 # File Library
 
-> Directus includes a full-featured Digital Asset Management (DAM) system for storing, organizing, browsing, and
-> transforming your various files and assets.
+> The File Library Module aggregates all files within the Directus Project into one consolidated library. It is a
+> full-featured Digital Asset Management (DAM) system for storing, organizing, browsing, and transforming your various
+> files and assets.
+
+![File Library](https://cdn.directus.io/docs/v9/app-guide/file-library/file-library-20220305A/file-library-20220305A.webp)
```
Comment (rijkvanzanten) [inline docs/app/file-library.md:7]: Can we use a screenshot without a folder named `ok` ?

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

**Comment by rijkvanzanten** [comment]
Other than that:

- Limits and controls for file sizes?

Not yet

- Are there any file types that cannot be imported?

There's no technical restrictions. That being said, only images / videos will be previewed in-app.

- Is it feasible to add an extension to upload Folders?

You can upload multiple files at a time already either through the upload flow in the video or through the api 

- Is cropping reversible? Or is this a destructive change? Idk how to reverse it!

No, file edits made will overwrite the original file on disk, and can't be reversed.

File: docs/app/file-library.md (hunk 2 above)
Comment (erondpowell) [reply docs/app/file-library.md:7]: lol yes we can boss!

File: docs/app/file-library.md (hunk 1 above)
Comment (erondpowell) [reply docs/app/file-library.md:214]: ok!

File: docs/app/file-library.md (hunk 2 above)
Comment (rijkvanzanten) [reply docs/app/file-library.md:7]: ok!

File: docs/app/file-library.md [hunk 3]
```diff
@@ -1,137 +1,209 @@
...
+::: tip Required Knowledge
 
-## Viewing or Editing a File
+Familiarity with the [Collections Page](/app/content-collections/) will be helpful but not necessary.
 
-![File Overview](https://cdn.directus.io/docs/v9/app-guide/file-library/viewing-or-editing-a-file-20220215A.webp)
+:::
+
+The File Library acts as one big [Folder](#folders) to store all uploaded [Files](#files). Sub-folders can be created
+within the File Library to help Files stay organized. Folders and Files are stored as Collections, which means
```
Comment (rijkvanzanten) [inline docs/app/file-library.md:27]:
```suggestion
within the File Library to help Files stay organized. Folders and Files information are stored in regular Collections, which means
```

File: docs/app/file-library.md [hunk 4]
```diff
@@ -1,137 +1,209 @@
...
-## Editing an Image
+::: tip Can't Delete Files Linked to Items
 
-Clicking the <span mi btn sec>tune</span> action button on the File Detail opens an image editor for rotating, cropping,
-or mirroring the file.
+Directus will not allow you to delete a File until you remove it from any and all related Items.
 
-1. From the **File Library**, click a file to open it's detail page
-2. Click the <span mi btn sec>tune</span> button in the top right to open the image editor
+:::
```
Comment (rijkvanzanten) [inline docs/app/file-library.md:75]: Lets update this, as it's not technically true. You can configure the relational constraint of your file field to SET NULL or CASCADE when the file is deleted. That way, the referencing field will get updated on the fly when the image is deleted.

Comment (rijkvanzanten) [inline docs/app/file-library.md:75]: Lets update this, as it's not technically true. You can configure the relational constraint of your file field to SET NULL or CASCADE when the file is deleted. That way, the referencing field will get updated on the fly when the image is deleted.

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]
Two last tiny nitpicks, good to merge after that ❇️

File: docs/app/file-library.md (hunk 4 above)
Comment (erondpowell) [reply docs/app/file-library.md:75]: Nice! Learn something new everyday

Comment (erondpowell) [reply docs/app/file-library.md]: This is a duplicate of the above comment.

Comment (erondpowell) [reply docs/app/file-library.md]: This should be resolved.

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**